### PR TITLE
Add docs for role.spec.allow.reason.mode

### PR DESCRIPTION
**Pull request by kopiczko**
Issue https://github.com/gravitational/teleport/issues/20164

### CODE REVIEW

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (r0mant) [inline docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:332]: Nit: A little shorter and still carries the point across.
```suggestion
resource allowed by "root-node-access" they will be required to provide a reason. If
a user's roleset includes multiple roles governing access requests to the same roles
and resources, "require" mode takes precedence.
```

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (r0mant) [inline docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:313]:
```suggestion
version: v7
```

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (r0mant) [inline docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:307]:
```suggestion
| `required` | The user must provide a non-empty reason when making a request. |
```

**Review by r0mant — APPROVED** [review]
lgtm but hold this off until after implementation is merged

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (kopiczko) [reply docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:313]: Should I change it in all examples in on this page?

Comment (r0mant) [reply docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:313]: Yeah role v7 is the latest, we should prefer using it.

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (nklaassen) [inline docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:300]: nit: "Reason mode" feels weird written out in english, maybe this is better
```suggestion
## Requiring request reasons

The `allow.request.reason.mode` field controls whether a reason is required when users submit access requests.
```

**Review by nklaassen — APPROVED** [review]

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (kopiczko) [reply docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:300]: I don't have a strong opinion here. @r0mant ?

Comment (r0mant) [reply docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx:300]: Nic's suggestion looks good to me.

**Review by smallinsky — APPROVED** [review]

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (kopiczko) [reply docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx]: 81b0dad5074ac2eb1c1f05c077722833ed5b655a

File: docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx
Comment (kopiczko) [reply docs/pages/admin-guides/access-controls/access-requests/access-request-configuration.mdx]: 14574945c841dd281e458f52c6c4486612bc12db

**Pull request merged**
